src/main/scala/carldata/series/Sessions.scala

```diff
@@ -25,25 +27,27 @@ object Sessions {
   }
 
   /** Finds a session with a duration tolerance
-    * @param maxTolerance the tolerance understood as the maximal duration between two samples n, n+k,
-    *                     where k>0, one of them is allowed to be 0, as for example:
-    *                       for Duration.ofSeconds(1)
+    * @param maxTolerance the tolerance between two samples in minutes, for example:
     *                       for a fragment of signal values distributed evenly for exactly one second
     *                       (2, 0, 1, 0, 0, 2, 3, 4, 5, 7, 0)
-    *                       there will be following sessions detected: [Session(0,2), Session(5, 9)]
+    *                       for Duration.ofSeconds(0) will be following sessions detected: [Session(0,0), Session(2,2), Session(5, 9)]
+    *                       for Duration.ofSeconds(1) will be following sessions detected: [Session(0,2), Session(5, 9)]
     * */
+
   def findSessions[V: Numeric](ts: TimeSeries[V], maxTolerance: Duration)(implicit num: Fractional[V]): Seq[Session] = {
     if (maxTolerance.compareTo(Duration.between(ts.head.get._1, ts.last.get._1)) >= 0)
       Seq(Session(0, ts.length - 1))
     else {
-      val xs = ts.rollingWindow(maxTolerance, sx => sx.sum)
-      val ys = xs.map(x =>
-        if (xs.slice(x._1, x._1.plusMinutes(maxTolerance.toMinutes).plusNanos(1)).values.contains(0))
-          num.fromInt(0)
+      val xs = Sessions.findSessions(ts)
+      val is = ts.index.zipWithIndex.map(x => (x._2, x._1))
```

Comment:
Those 2 operations are not needed. You can use ts.index to access DateTime at given index.